### ISSUE TITLE
❇ forked-daapd

### ISSUE THREAD

**Issue by gerard33** [hassio-addons/repository#98]
## Proposed new add-on
I saw the new card by @kalkih and forked-daapd seems like a nice addition for the Hassio addons.

http://ejurgensen.github.io/forked-daapd/
https://community.home-assistant.io/t/lovelace-forked-daapd-card/72112

<blockquote><div>forked-daapd</div><div><strong><a href="http://ejurgensen.github.io/forked-daapd/">forked-daapd</a></strong></div><div>Linux/FreeBSD DAAP (iTunes) and MPD media server with support for AirPlay devices (multiroom), Apple Remote (and compatibles), Chromecast, Spotify and internet radio.</div></blockquote>
<blockquote><img src="https://assets-cdn.github.com/favicon.ico" width="48" align="right"><div><img src="https://community.home-assistant.io/uploads/default/original/1X/510301398ed5188e395eafb0983fbf8fe8ffcbcf.png" height="14"> Home Assistant Community</div><div><strong><a href="https://community.home-assistant.io/t/lovelace-forked-daapd-card/72112">Lovelace: Forked daapd card</a></strong></div><div>Lovelace: forked daapd card This card is very niche, I made it mostly to cover my own needs, but thought I would share it anyway, hopefully someone can find it useful!    A lovelace card for Home Assistant to control a forked daapd instance.  This card allow you to control your forked daapd instance through Home Assistant.  The following controls are available: master volume, play/pause, next/previous, on/off, output selection, individual output volume.    Prerequisite  Forked daapd built with w...</div></blockquote>

**Comment by postlund** [comment]
Any progress on this? Would LOVE to see this happen! 👍

**Comment by frenck** [comment]
Yes, it is partially done.
It runs and mostly works correctly. I still have an issue with the websocket connection left, which I'm currently trying to figure out.

**Comment by postlund** [comment]
Cool! I had some issues with websockets as well (mostly the same as in https://github.com/linuxserver/docker-daapd/issues/23 and https://github.com/linuxserver/docker-daapd/issues/27), is it the same thing you have

**Comment by lancelon** [comment]
So cool, Frenck! I think this is the key to "say" working - getting Forked-Daapd on the same IP range and subnet etc as Hass?

**Comment by frenck** [comment]
@lancelon Your reply does not make sense to me. What are you trying to tell me?

**Comment by nicx** [comment]
@frenck any news to this? would be really great to see this shortly 👍

**Comment by frenck** [comment]
Unfortunately not. Progress is really slow, at a point where I'm leaning towards saying it might not happen at all.

**Comment by lancelon** [comment]
Was it the websockets issue?

**Comment by thies226j** [comment]
how is development going?

**Comment by frenck** [comment]
It is not going. I cannot fix the upstream project.

**Comment by lancelon** [comment]
Howdy frenck - is it the websockets issue you mentioned earlier? Is forked-daapd doing something "non standard" that means you can't interact with it properly?
Cheers

**Comment by lancelon** [comment]
Mr Frenck! Just wondering if I could help talk to the developers over at forked-daapd?

**Comment by Hackashaq666** [comment]
FYI, it looks like there have been significant updates to
https://github.com/linuxserver/docker-daapd
I was able to get it running on Home Assistant (formerly Hassio) using the Portainer addon.
May be worth revisiting if anyone is still interested in this.

**Comment by frenck** [comment]
I'll take a look, the add-on is fully build, just that issue was left. 

Will upgrade the add-on and see how it goes.

**Comment by frenck** [comment]
Thanks for letting know btw :)

**Comment by SilvrrGIT** [comment]
Any progress on this add-on?  I would love to run this on my HomeAssistant box rather than another box.  Thanks!

**Comment by ddppddpp** [comment]
I spent a night hacking at [johnpdowling's](https://github.com/johnpdowling) [forked-daapd add-on](https://github.com/johnpdowling/hassio-addons/tree/master/forked-daapd) build using frenck's SSH add-on as a guide and finally managed to build and run it on amd64 and armv7l. The code is quite ugly and will probably remain so, but if you want can have a [look](https://gist.github.com/ddppddpp/a453fff65d53044cd0ec7cd3d0afd814). Working with no issues so far but I have no time to test i.e. tts stuff and multi-room syncing.

**Comment by staraxis** [comment]
Hi @ddppddpp what would be the best way to test this? Should I use the gist to create my own addon for testing purposes. Happy to provide feedback for this

**Comment by Ulrar** [comment]
So I got it to work, built on a debian base to get libspotify.
Websockets do work, but I had to set it up in an external HAProxy with SSL to resolve insecure ws:// calls. I imagine what frenck meant with the websockets issue is trying to get that to work directly through hassio, without needing an external service, as this is not very user friendly.

The only thing I couldn't get to work is the web interface in ingress, looks like the player doesn't like iframes, but using port 3689 directly does work for that.
Again I'll probably get around it through my external haproxy later on, at least for now.

And finally to get the forked daapd card to work in the frontend I had to edit the javascript, as it seems like that card has ws:// and http:// hardcoded in it. But after all that the card does work fine, listing all the outputs with individual controls for each speakers and so on, with no latency, very neat.

**tl;dr** this is a lot of fiddling and pretty dirty, but it does run, you can find the Dockerfile [here](https://gist.github.com/Ulrar/60f29c58f3f5cad0758a6728c394e675).
Sorry about the list of apt-get installs, I was trying stuff out for a while. I'll need to clean that up and remove build dependencies from the final image at some point.
You'll also need to start dbus and avahi-daemon in your services.d for this to start correctly.

**Comment by Ulrar** [comment]
Alright so I got it to work pretty well, the only thing missing is ingress, and exposing the configs in the hassio interface.
If anyone is interested I put the addon in a repo : https://github.com/Ulrar/hassio-addons

I've tested airplay, chromecast, spotify, DAAPD and Remote clients, they all work great.
It gets detected by Home Assistant as a new integration and that creates a media player for the server and one for each of the connected speakers (chromecast, airplay, pulseaudio ..), exactly as you'd expect.
I'd advise using the mini media player card rather than the forked-daapd card however, much more reliable since it doesn't have websocket issues.

**Comment by christian1986** [comment]
Any news here:)?
